extmod/modbluetooth.c

```diff
@@ -143,6 +147,26 @@ STATIC mp_obj_t bluetooth_uuid_unary_op(mp_unary_op_t op, mp_obj_t self_in) {
     }
 }
 
+STATIC mp_obj_t bluetooth_uuid_binary_op(mp_binary_op_t op, mp_obj_t lhs_in, mp_obj_t rhs_in) {
+    mp_obj_bluetooth_uuid_t *lhs = MP_OBJ_TO_PTR(lhs_in);
+    mp_obj_bluetooth_uuid_t *rhs = MP_OBJ_TO_PTR(rhs_in);
+    switch (op) {
+        case MP_BINARY_OP_EQUAL:
+        case MP_BINARY_OP_LESS:
+        case MP_BINARY_OP_LESS_EQUAL:
+        case MP_BINARY_OP_MORE:
+        case MP_BINARY_OP_MORE_EQUAL:
+            if (lhs->type == rhs->type) {
+                return mp_obj_new_bool(mp_seq_cmp_bytes(op, lhs->data, lhs->type, rhs->data, rhs->type));
+            } else {
+                return mp_binary_op(op, MP_OBJ_NEW_SMALL_INT(lhs->type), MP_OBJ_NEW_SMALL_INT(rhs->type));
```

Comment:
So if the types (length of UUID data) are different then the shorter one (eg 2 bytes vs 4 bytes) is considered smaller?  I guess that makes the most sense, is the most obvious way to do it.  Other way might be to pad the shorter value with zeros to make it the same size, but that might not give correct semantics.

Reply:
I don't think it's meaningful to directly compare two UUIDs of different length, so the aim was to provide something that still maintains a consistent ordering behavior with minimal code size.
Comparing UUIDs of different length might be relevant in the context of comparing prefixes/suffixes or something, but it would be better to be explicit.

I've fixed the other comments. Thanks